tests/foreman/cli/test_product.py

```diff
@@ -36,274 +36,278 @@
-
-        :expectedresults: Product is not created
-
-        :CaseImportance: High
-        """
-        product_name = gen_alphanumeric()
-        for invalid_label in (
-            gen_string('latin1', 15),
-            gen_string('utf8', 15),
-            gen_string('html', 15),
-        ):
-            with self.subTest(invalid_label):
-                with self.assertRaises(CLIFactoryError):
-                    make_product(
-                        {
-                            'label': invalid_label,
-                            'name': product_name,
-                            'organization-id': self.org['id'],
-                        }
-                    )
-
-    @pytest.mark.run_in_one_thread
-    @pytest.mark.tier2
-    @pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
-    def test_product_list_with_default_settings(self):
-        """Listing product of an organization apart from default organization using hammer
-         does not return output if a defaults settings are applied on org.
-
-        :id: d5c5edac-b19c-4277-92fe-28d9b9fa43ef
-
-        :BZ: 1745575
-
-        :expectedresults: product/reporsitory list should work as expected.
-
-        """
-        default_product_name = gen_string('alpha')
-        non_default_product_name = gen_string('alpha')
-        default_org = self.org
-        non_default_org = make_org()
-        default_product = make_product(
-            {'name': default_product_name, 'organization-id': default_org['id']}
-        )
-        non_default_product = make_product(
-            {'name': non_default_product_name, 'organization-id': non_default_org['id']}
+                'organization-id': org_id,
+                'product-id': product['id'],
+                'url': FAKE_0_YUM_REPO,
+            },
         )
-        for product in (default_product, non_default_product):
-            make_repository({'product-id': product['id'], 'url': FAKE_0_YUM_REPO})
 
-        Defaults.add({'param-name': 'organization_id', 'param-value': default_org['id']})
+    Defaults.add({'param-name': 'organization_id', 'param-value': org_id})
+    result = ssh.command('hammer defaults list')
+    assert org_id in "".join(result.stdout)
+    try:
+        # Verify --organization-id is not required to pass if defaults are set
+        result = ssh.command('hammer product list')
+        assert default_product_name in "".join(result.stdout)
+        result = ssh.command('hammer repository list')
+        assert default_product_name in "".join(result.stdout)
+
+        # verify that defaults setting should not affect other entities
+        product_list = Product.list({'organization-id': non_default_org['id']})
+        assert non_default_product_name == product_list[0]['name']
+        repository_list = Repository.list({'organization-id': non_default_org['id']})
+        assert non_default_product_name == repository_list[0]['product']
+
+    finally:
```

Comment:
This is just my suggestion, I think we can achieve it without using try and finally like below.
```
def __cleanup(org_id):
    Defaults.delete({'param-name': 'organization_id'})
    result = ssh.command('hammer defaults list')
    assert org_id in "".join(result.stdout) 
```
and call request.addfinalizer(lambda: __cleanup(org_id)) in the begining of the test case.

Reply:
I left the original `try / finally` in place, because the `assert` after removing the default parameters should be part of the test.